config/routes.rb

```diff
@@ -0,0 +1,20 @@
+# For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+Rails.application.routes.draw do
+
+  root 'works#main'
+
+  resources :works do
+    resources :votes, only: [:create]
+  end
+
+  # TODO Write this nested route - if you are not logged in you cannot Upvote
+  get '/works/:work_id/upvote', to: 'votes#create', as: 'upvote'
+
+  resources :users
+
```

Comment:
I don't think you need all 7 restful routes for `users` - only `index` and `show` are needed for this project.